packages/gatsby-theme-docs/gatsby-node.js

```diff
@@ -174,32 +177,61 @@ function generateReleaseNoteSlug(node) {
 
 // https://www.gatsbyjs.org/docs/mdx/programmatically-creating-pages/#create-pages-from-sourced-mdx-files
 exports.createPages = async ({ graphql, actions, reporter }, pluginOptions) => {
-  const allContentPagesResult = await graphql(`
-    query QueryAllContentPages {
-      contents: allFile(
-        filter: {
-          sourceInstanceName: { eq: "content" }
-          internal: { mediaType: { eq: "text/mdx" } }
+  const allMdxPagesResult = await graphql(
+    `
+      query QueryAllContentPages($hasReleaseNotes: Boolean!) {
+        contents: allFile(
+          filter: {
+            sourceInstanceName: { eq: "content" }
+            internal: { mediaType: { eq: "text/mdx" } }
+          }
+        ) {
+          nodes {
+            childMdx {
+              id
+              fields {
+                ...MdxPageFields
+              }
+            }
+            name
+          }
         }
-      ) {
-        nodes {
-          childMdx {
-            id
-            fields {
-              slug
-              title
-              beta
-              isGlobalBeta
-              excludeFromSearchIndex
+        releaseNotes: allFile(
+          filter: {
+            sourceInstanceName: { eq: "releases" }
+            internal: { mediaType: { eq: "text/mdx" } }
+          }
+        ) @include(if: $hasReleaseNotes) {
```

Comment:
@davifantasia this is what I meant with the conditional directive. This way we still send 1 graphql query and we can use fragments.

Reply:
Damn, it does not work. I guess gatsby parses the schema first and it fails at that point.